oaknational/Oak-Web-Application
# Feat/lesq 291/hamburger update [LESQ-291]

## Description
Ended up being a few changes, key ones to review are:
- I added `early-realease-units` to `urls.ts` in order to be able to link to that page from the hamburger menu - I'm not sure if this has any ramifications or should have been done differently
- created new types for the new menu links rather than try and shoe horn into the existing ones
- created new components for the hamburger menu sections and links, imaginatively called `NewMenuLink` and `NewMenuSections` - better suggestions on a postcard
- replaces the `betaMenuSections` with updated links and replaced them in `AppHeader` - I believe this means it won't be visible on the old page and only the new ones so correct me if wrong
- updated some styles

For Monika:
- I couldn't find an existing button that matched the style in the design for `hover` and `focus` exactly, I used `ButtonWithIcon` in the variant `buttonStyledAsLink`, but could alternately use the`minimal` variant which[ looks like this ](https://storybook.thenational.academy/?path=/story/buttons-button--button-with-icon/path=/story/buttons-button--button-with-icon&args=label:Home;variant:minimal) but with the icon at the end.

## How to test

1. Go to https://deploy-preview-1908--oak-web-application.netlify.thenational.academy/teachers-2023
2. Click on the hamburger menu
3. You should see the new design and all the links should go to the correct place

## Screenshots

How it used to look (delete if n/a):
<img width="500" alt="Screenshot 2023-09-15 at 12 19 51" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/775a2f91-a06e-40dd-8aa5-923f45fca735">
  
  
How it should now look:
<img width="500" alt="Screenshot 2023-09-15 at 12 20 46" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/67addc89-5b1c-416f-ae30-e0a90ce4999a">

## ACs
- [ ]  Scrolls if it doesn’t fit
- [ ]  New links
    - [ ]  Hover
    - [ ]  Focus
- [ ]  “New” next to early release units
- [ ]  Opens in a new tab icon on some of the links
- [ ]  New “X” logo for Elon Musk
